src/developer-quickstart.md

```diff
@@ -0,0 +1,482 @@
+
+A few pieces of info that will be helpful before moving on:
+
+- This guide was created using VSCode as the code editor.
+- Download the Sway language extension in VSCode to get syntax highlighting, keyboard shortcuts, and more.
+- Download the rust-analyzer extension in VSCode to get syntax highlighting, code completion, and more.
+
+**Start by creating a new, empty folder. We'll call it `fuel-project`.**
+
+### Writing the Contract
+
+First, let's [install the Sway toolchain](https://github.com/FuelLabs/fuelup).
+
+Then with `forc` installed, create a contract project inside of your `fuel-project` folder:
+
+```sh
+cd fuel-project
+forc new counter_contract
+```
+
+Here is the project that `Forc` has initialized:
+
+```console
+├── fuel-project
+    ├── Cargo.toml
+    ├── Forc.toml
+    ├── src
+    │   └── main.sw
+    └── tests
+        └── harness.rs
+```
+
+`Forc.toml` is the _manifest file_ (similar to `Cargo.toml` for Cargo or `package.json` for Node) and defines project metadata such as the project name and dependencies. You'll notice a `cargo.toml` file because `forc` uses `cargo` under the hood.
+
+Open your project in a code editor and delete the boilerplate code in `src/main.sw` so that you start with an empty file.
+
+Every Sway file must start with a declaration of what type of program the file contains; here, we've declared that this file is a contract.
+
+```sway
+contract;
+```
+
+Next, we'll define a storage value. In our case, we have a single counter that we'll call `counter` of type 64-bit unsigned integer and initialize it to 0.
+
+```sway
+storage {
+    counter: u64 = 0,
+}
+```
+
+### ABI
+
+An ABI defines an interface, and there is no function body in the ABI. A contract must either define or import an ABI declaration and implement it. It is considered best practice to define your ABI in a separate library and import it into your contract because this allows callers of the contract to import and use the ABI in scripts to call your contract.
+
+For simplicity, we will define the ABI natively in the contract.
+
+```sway
+abi Counter {
+    #[storage(read, write)]
+    fn increment();
+
+    #[storage(read)]
+    fn count() -> u64; 
+}
+```
+
+### Going line by line
+
+`#[storage(read, write)]` is an annotation which denotes that this function has permission to read and write a value in storage.
+
+`fn increment();` - We're introducing the functionality to increment and denoting it shouldn't return any value.
+
+`#[storage(read)]` is an annotation which denotes that this function has permission to read values in storage.
+
+`fn counter() -> u64;` - We're introducing the functionality to increment the counter and denoting the function's return value.
+
+### Implement ABI
+
+Below your ABI definition, you will write the implementation of the functions defined in your ABI.
+
+```sway
+impl Counter for Contract {
+    #[storage(read)]
+    fn count() -> u64 {
+      return storage.counter;
```

Comment:
It's more idiomatic to use an implicit return, i.e. `storage.counter`. Please change explicit tail returns to implicit returns.